mm2src/mm2_main/src/lp_healthcheck.rs

```diff
@@ -52,14 +53,23 @@ impl HealthcheckMessage {
         Ok(Self { signature, data })
     }
 
-    pub(crate) fn is_received_message_valid(&self, my_peer_id: PeerId) -> bool {
+    pub(crate) fn is_received_message_valid(&self, my_peer_id: PeerId, healthcheck_config: &HealthcheckConfig) -> bool {
         let now = Utc::now().timestamp();
-        if now > self.data.expires_at {
+        let remaining_expiration_seconds = u64::try_from(self.data.expires_at - now).unwrap_or(0);
+
+        if remaining_expiration_seconds == 0 {
             log::debug!(
                 "Healthcheck message is expired. Current time in UTC: {now}, healthcheck `expires_at` in UTC: {}",
                 self.data.expires_at
             );
             return false;
+        } else if remaining_expiration_seconds > healthcheck_config.message_expiration_secs {
```

Comment:
i think it makes more sense to check `self.data.expires_at` and not `remaining_expiration_time`.

Reply:
Can you clarify what do you mean here?

Reply:
what if clocks on the requester and responder are unsynchronised?

Reply:
> what if clocks on the requester and responder are unsynchronised?

That's general problem for whole KDF logic including swaps and we have solution for that (we discussed that topic in the internal team channel a week ago or so).

Reply:
> > what if clocks on the requester and responder are unsynchronised?
> 
> That's general problem for whole KDF logic including swaps and we have a [solution]

Currently allowed difference is 60 sec.
This PR's code relies on that in future there will be no difference at all, right?

Reply:
Doesn't need to be precise that much. That's why we use 10 seconds as a default expiration time. 2-3 seconds time gap will be fine now and in the future. But yeah, this implementation relies on the fact that any peer in the network will not have unsynced (e.g., by more than 4-5 seconds) time.